qiskit/qpy/binary_io/circuits.py

```diff
@@ -1070,12 +1121,55 @@ def _write_pauli_evolution_gate(file_obj, evolution_gate, version):
     def _write_elem(buffer, op):
         elem_data = common.data_to_binary(op.to_list(array=True), np.save)
         elem_metadata = struct.pack(formats.SPARSE_PAULI_OP_LIST_ELEM_PACK, len(elem_data))
+        if version >= 17:
+            elem_sparse_operator = struct.pack("?", False)
+            buffer.write(elem_sparse_operator)
         buffer.write(elem_metadata)
         buffer.write(elem_data)
 
+    def _write_elem_sparse(buffer, op):
+        bitterms = op.bit_terms
+        coeffs = op.coeffs
+        bounds = op.boundaries
+        inds = op.indices
+        numq = op.num_qubits
+
+        coeff_real = []
+        coeff_imag = []
+        for coeff in coeffs:
+            coeff_real.append(coeff.real)
+            coeff_imag.append(coeff.imag)
+        coeff_data = struct.pack(
+            f"{len(coeffs)*2}d", *(val for pair in zip(coeff_real, coeff_imag) for val in pair)
+        )
```

Comment:
We should be able to do this without creating an intermediary list, e.g.
```suggestion
        # pack elements as [c1.real, c1.imag, c2.real, c2.imag, ...]
        coeff_data = struct.pack(
            f"{len(coeffs)*2}d", *(val for coeff in coeffs for val in (coeff.real, coeff.imag))
        )
```

Reply:
Thanks. :) Committed suggestion.